src/ASM/LR/ASMu3D.C

```diff
@@ -2029,3 +2118,71 @@ void ASMu3D::generateThreadGroups (const Integrand& integrand, bool silence,
     std::cout << ": "<< threadGroups[0][i].size() <<" elements";
   }
 }
+
+
+bool ASMu3D::updateDirichlet (const std::map<int,RealFunc*>& func,
+                              const std::map<int,VecFunc*>& vfunc, double time,
+                              const std::map<int,int>* g2l)
+{
+
+  std::map<int,RealFunc*>::const_iterator    fit;
+  std::map<int,VecFunc*>::const_iterator     vfit;
+  std::map<int,int>::const_iterator          nit;
+
+  for (size_t i = 0; i < dirich.size(); i++)
+  {
+    // figure out function index offset (when using multiple basis)
+    size_t ofs = 0;
+    for (int j = 1; j < dirich[i].basis; j++)
+      ofs += this->getNoNodes(j);
+
+    RealArray edgeControlpoints;
+    Real2DMat edgeControlmatrix;
+    if ((fit = func.find(dirich[i].code)) != func.end())
+      this->faceL2projection(dirich[i], *fit->second, edgeControlpoints, time);
+    else if ((vfit = vfunc.find(dirich[i].code)) != vfunc.end())
+      this->faceL2projection(dirich[i], *vfit->second, edgeControlmatrix, time);
+    else
+    {
+      std::cerr <<" *** ASMu3D::updateDirichlet: Code "<< dirich[i].code
+                <<" is not associated with any function."<< std::endl;
+      return false;
+    }
+    if (edgeControlpoints.empty() && edgeControlmatrix.empty())
+    {
+      std::cerr <<" *** ASMu3D::updateDirichlet: Projection failure."
+                << std::endl;
+      return false;
+    }
+
+    // Loop over the nodes of this boundary curve
+    for (nit = dirich[i].MLGN.begin(); nit != dirich[i].MLGN.end(); nit++)
```

Comment:
And this loop becomes: `j = 0; for (int node : dirich[i].MLGN)`
and substitute `nit->second` by `node` and `nit->first`by `j` below (and increment j properly since can't do that in the range-based loop itself).

Totally up to you if you want to do it or not, since it touches some lines, but in my opinion it increases the readability.